### PR TITLE
Add unit tests for SA1004

### LINKED ISSUE #229: Diagnostic unit tests: SA1004 (DocumentationLinesMustBeginWithSingleSpace)
Implement unit tests for #10.

### PR DESCRIPTION
**Pull request by sharwell**
This diagnostic is also enabled by default now.

Fixes #229

### CODE REVIEW

**Comment by pdelvo** [comment]
Should we also report warnings for multiline xml comments?

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/SpacingRules/SA1004UnitTests.cs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    ///</summary>
+    ///   <param name=""x"">The document root.</param>
+    ///    <param name=""y"">The XML header token.</param>
+    private void Method1(int x, int y)
+    {
+    }
+}
+";
+
+            // Currently the extra indentation for <param> elements is not checked.
```
Comment (pdelvo) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/SpacingRules/SA1004UnitTests.cs]: We should probably add a skipped test for that.

Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/SpacingRules/SA1004UnitTests.cs]: Do we *want* to check for extra indentation? It caused a problem in the past (#715). If you think we should check extra indentation for some set of rules, I'm thinking we should create a new issue to define exactly what those rules are.

**Comment by sharwell** [comment]
> Should we also report warnings for multiline xml comments?

Probably, but I'm thinking that should be a separate issue. It wouldn't be the first time we implemented support for single-line documentation comments but left out the multi-line version.

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/SpacingRules/SA1004UnitTests.cs (hunk 1 above)
Comment (pdelvo) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/SpacingRules/SA1004UnitTests.cs]: Sounds good

**Comment by pdelvo** [comment]
Im okay with not supporting multiline documentation for now. I have not seen anyone using them

**Pull request merged**